packages/xo-server/src/xo-mixins/rest-api.mjs

```diff
@@ -1514,6 +1514,15 @@ export default class RestApi {
         res.sendStatus(200)
       })
     )
+
+    api.delete(
+      '/:collection(groups)/:id',
+      wrap(async (req, res) => {
+        const { id } = req.params
+        await app.deleteGroup(id)
+        res.sendStatus(204)
+      })
+    )
```

Comment:
`app.deleteGroup` can throw, so this needs to be handled, otherwise the request will return an internal server error, and will not help the end user.
Something like this should work and be enough for now: 
```js
        try {
          await app.deleteGroup(id)
          res.status(204) // HTTP status 204  -> success
        } catch (error) {
          res.json({ error: error.message })
          if (noSuchObject.is(error)) {
            res.status(404) // HTTP status 404 -> resource not found (so the group with the given id was not found)
          } else {
            res.status(500) // HTTP status 500 -> internal server error (unhandled error in this case)
          }
        }
        res.send()
```